subsys/pm/device.c

```diff
@@ -119,6 +119,13 @@ int pm_device_action_run(const struct device *dev,
 
 		state = PM_DEVICE_STATE_OFF;
 		break;
+	case PM_DEVICE_ACTION_TURN_ON:
+		if (pm->state != PM_DEVICE_STATE_OFF) {
+			return -EALREADY;
+		}
+
+		state = PM_DEVICE_STATE_SUSPENDED;
```

Comment:
note: according to docs, not implementing support for TURN_ON should not have any implications, but it has as it is now: action_cb will return -ENOTSUP and so suspended state will not be applied (see L134)

Reply:
tldr; we need some unit tests that check API behavior

Reply:
same thing for `TURN_OFF` now that is used by a power domain to notify children.